cmd/cosign/cli/generate_key_pair.go

```diff
@@ -82,6 +85,31 @@ func GenerateKeyPairCmd(ctx context.Context, kmsVal string) error {
 		}
 		fmt.Fprintln(os.Stderr, "Public key written to cosign.pub")
 		return nil
+	} else if ykVal {
+		yk, err := cosign.GetYubikey()
+		if err != nil {
+			return err
+		}
+		defer yk.Close()
+
+		pk, err := cosign.GenYubikey(yk)
+		if err != nil {
+			return err
+		}
+
+		der, err := x509.MarshalPKIXPublicKey(pk)
+		if err != nil {
+			return err
+		}
+		pemBytes := pem.EncodeToMemory(&pem.Block{
+			Type:  "PUBLIC KEY",
+			Bytes: der,
+		})
+		if err := ioutil.WriteFile("cosign.pub", pemBytes, 0600); err != nil {
+			return err
+		}
+		fmt.Fprintln(os.Stderr, "Public key written to cosign.pub")
```

Comment:
this goes against the cmd help message which says ("generates key pair") but it generates only a .pub here. not sure what to do with it.

Reply:
It's a little subtle I guess. It does generate a key pair, but the only file it writes to disk is the public key. The private one is generated on the key and cannot be exported.

Reply:
I improved the output here a bit.

There is still A LOT to be done here :)